### PR TITLE
Oceanstor file driver

### PR DESCRIPTION
**Pull request by jackhaibo**
**What this PR does / why we need it**:
Implementation of `oceanstor `file share driver. All basic functions are completed, waiting to dock with the dock.

**Special notes for your reviewer**:
@Shruthi-1MN @xxwjj, @wisererik, @leonwanghui, @xing-yang , @NajmudheenCT, @ kumarashit , @rhsakarpos, @himanshuvar, @skdwriting @PravinRanjan10

### CODE REVIEW

**Comment by skdwriting** [comment]
@jackhaibo, please rebase and update the PR.

File: contrib/drivers/filesharedrivers/oceanstor/oceanclient.go [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+func (c *Client) deleteFileSystem(name, poolID string) error {
+	data := map[string]interface{}{
+		"PARENTID":   poolID,
+		"NAME":       name,
+		"PARENTTYPE": 216,
+		"ALLOCTYPE":  1,
+	}
+
+	url := "/filesystem"
+	resp, err := c.request(url, "POST", data)
```
Comment (leonwanghui) [inline contrib/drivers/filesharedrivers/oceanstor/oceanclient.go]: Why it's `POST` method here?

File: contrib/drivers/filesharedrivers/oceanstor/oceanclient.go [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+	if err := json.Unmarshal(resp, &fsSnapshot); err != nil {
+		return nil, err
+	}
+
+	return &fsSnapshot.Data, nil
+}
+
+func (c *Client) getAllFilesystem() ([]FileSystemData, error) {
+	url := "/filesystem"
+	resp, err := c.request(url, "Get", nil)
```
Comment (leonwanghui) [inline contrib/drivers/filesharedrivers/oceanstor/oceanclient.go]: s/Get/GET

File: contrib/drivers/filesharedrivers/oceanstor/oceanstor.go [hunk 3]
```diff
@@ -0,0 +1,702 @@
...
+		return errors.New(msg)
+	}
+
+	d.Client = cli
+	log.Info("get oceanstor client successfully")
+
+	return nil
+}
+
+func (d *Driver) IniConf() {
```
Comment (leonwanghui) [inline contrib/drivers/filesharedrivers/oceanstor/oceanstor.go]: Please rename this method to `InitConf`.

File: contrib/drivers/filesharedrivers/oceanstor/util_test.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+// limitations under the License.
+
+package oceanstor
+
+import (
+	"reflect"
+	"testing"
+)
+
+var assertTestResult = func(t *testing.T, got, expected interface{}) {
```
Comment (leonwanghui) [inline contrib/drivers/filesharedrivers/oceanstor/util_test.go]: Please add more test cases for other methods in util.go.

File: contrib/drivers/filesharedrivers/oceanstor/oceanclient.go (hunk 1 above)
Comment (jackhaibo) [reply contrib/drivers/filesharedrivers/oceanstor/oceanclient.go]: This method has been deprecated and will be deleted.

File: contrib/drivers/filesharedrivers/oceanstor/oceanclient.go (hunk 2 above)
Comment (jackhaibo) [reply contrib/drivers/filesharedrivers/oceanstor/oceanclient.go]: Done. In the `c.request` method, the `GET`, `UPDATE` and so on are all changed to uppercase with strings.ToUpper(). The method name in this case has no effect.

File: contrib/drivers/filesharedrivers/oceanstor/oceanstor.go (hunk 3 above)
Comment (jackhaibo) [reply contrib/drivers/filesharedrivers/oceanstor/oceanstor.go]: Done.

File: contrib/drivers/filesharedrivers/oceanstor/util_test.go (hunk 4 above)
Comment (jackhaibo) [reply contrib/drivers/filesharedrivers/oceanstor/util_test.go]: Done.

**Pull request merged**